src/api/liveSpecsExt.ts

```diff
@@ -0,0 +1,127 @@
+import {
+    CONNECTOR_IMAGE,
+    CONNECTOR_TITLE,
+    defaultTableFilter,
+    QUERY_PARAM_CONNECTOR_TITLE,
+    SortingProps,
+    supabaseClient,
+    TABLES,
+} from 'services/supabase';
+import { CatalogStats, LiveSpecsExtBaseQuery } from 'types';
+
+const baseColumns = [
+    'catalog_name',
+    'id',
+    'spec_type',
+    'updated_at',
+    'last_pub_id',
+];
+
+const commonColumns = baseColumns.concat([
+    'connector_id',
+    'connector_image_name',
+    'connector_image_tag',
+    CONNECTOR_IMAGE,
+    CONNECTOR_TITLE,
+    'writes_to',
+]);
+
+export interface CaptureQuery extends LiveSpecsExtBaseQuery {
+    writes_to: string[];
+}
+export interface CaptureQueryWithStats extends CaptureQuery {
+    stats?: CatalogStats;
+}
+export interface MaterializationQuery extends LiveSpecsExtBaseQuery {
+    reads_from: string[];
+}
+export interface MaterializationQueryWithStats extends MaterializationQuery {
+    stats?: CatalogStats;
+}
+export type CollectionQuery = Pick<
```

Comment:
Query typing has been on my mind pretty heavily, particularly because of the constraints they can impose on segments of a store state. The `spec` and `setSpecs` slices of the editor store state are a good example of a _query type constraint_ place on a store. Here, it is important that there is a `draft_specs` query type that is an analog of a `live_specs` query. This is easy enough to achieve, but as we extend/prune query types and apply them to query helper functions, utility functions, and store assets, it may prove beneficial to have a _specific_ approach for query type generation. An idea periodically comes to mind: create a base type that identifies the structure of an entire table and utilize the `Pick` utility to create common subsets of that type. _Thinking out loud._

Reply:
Yeah - I think once we get the Supabase CLI updated and have it type the DB this will be a lot easier. As we'll always be starting with "a base type" of the table. 

I agree with the thinking out loud - as I was coding like this for this PR it felt alright and easy to work with.